torch_brain/transforms/binning.py

```
@@ -0,0 +1,84 @@
+        AttributeError: If any component of the path cannot be resolved.
+    """
+    # Split key by dots, resolve using getattr
+    components = path.split(".")
+    obj = data
+    for c in components[:-1]:
+        try:
+            obj = getattr(obj, c)
+        except AttributeError:
+            raise AttributeError(
+                f"Could not resolve {path} in data (specifically, at level {c})"
+            )
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)